### ISSUE TITLE
Mounts Page images are all off

### ISSUE THREAD

**Issue by bTackt** [HabitRPG/habitica#4618]
As I presume it was mostly skinned from the pets page, the mounts page is off. See here:

![screenshot_1](https://cloud.githubusercontent.com/assets/1434020/6026645/7e4ec112-abaa-11e4-9d31-579596045332.png)

In the red, I write how many mounts I have unlocked in each row. Due to the head sprites being significantly larger than pets, the corresponding black paw prints are not the same size. There needs to be uniformity in the head sprite size and black paw prints (I don't know if each type of mount has the same width/height though, I didn't check), otherwise, this ghastly (gasp! :P) misalignment occurs.

Solution proposed: enlarge pawprints to a "mount size" for this page.

Also, it has been mentioned in #3397, but only using the head sprites is a bit disconcerting. Not only is it off center due to the inherent nature of the sprites, some mounts look disfigured (the dragons, for example, I believe, have an addition wing). This also makes one need to click on each to test out the effect of the appearance of the avatar; you can't just look and choose, like pets. It is also to be noted that since yall are fixing things like sidesaddling problems, the head sprites for wolves, at least, is going to look extra bizarre.

Solution proposed: Either use both head and body sprites together, or have another sprite sheet (this doesn't take too much effort/space, I presume...right?) with just heads (no wings, half-wolf bodies, etc.) which are centered.

Of course, the heads-only solution is much more complicated, as all the extra mounts other than the 90 would need redos. Head and body is a better solution, but in #3397 Alys mentioned something about creating new images (but wait, sprites don't create new images...you just position them with CSS? perhaps I'm wrong. Can you not load a singular sprite sheet image into a page more than once? sorry rusty HTML). In addition, one would have to worry about total mount size being uniform to keep the grid format (which I'm not sure if they're uniform; for one thing, wolves just feel like they're much longer horizontally).

^sorry long and possibly complicated ramble
##

**Comment by crookedneighbor** [comment]
I'm glad I'm not the only one that is bugged by this.

I don't think displaying the full mount is a good idea, mostly because of the amount of real estate it would take up. Though, maybe we could add the full mount image in the popover that displays the name. That could be cool!

@lemoness What do you think about having separate head only images of the mounts? I'd like to make these all a consistent size, and the dragon and flying pig wings have always bugged me. Now, with the new mount images to fix the side-saddling look, it'll look like we cut those mounts in two using the current method.

Isolating the heads for the few mounts that require it is simple enough pixel work that I think I could make the adjustments. It'd just be a matter of requiring future mount art contributions to include a head only version of the image (which in many cases will likely be a duplicate file cropped to a specific size and centered). 

Obviously, going that route would add a large number of extra sprites. At what point do we split to 7 spritesheets instead of 6?

**Comment by lemoness** [comment]
So here's the thing - we've already accepted a ton of mounts to the queue.
Right now there are 178 pending individual mounts (not species; individual
files) that have the art submitted but are not in-game, in addition to the
ones that are in-game. So.... we can do this, but it's going to take some
pixel-pushing. If you decide you really want to do this, I can make a new
Trello card for artists to do it and count it towards their contributor
tiers.

**Comment by Alys** [comment]
@knittingfrenzy18 _" in #3397 Alys mentioned something about creating new images (but wait, sprites don't create new images...you just position them with CSS?"_ No, I said we might **_not**_ need to create new images because we might be able to use CSS trickery instead. :) https://github.com/HabitRPG/habitrpg/issues/3397#issuecomment-53149901 I don't know much about how sprites work, so this might be impossible, but I was thinking we could just specify different offsets for each mount to display only the head portion, without needing head-only images.

**Comment by crookedneighbor** [comment]
@Alys I don't think that method would be impossible, but you'd have to do all kinds of crazy routing with the css templating in the grunt task and it would have to be specialized for each set of heads. Personally, I don't think it's worth it. Plus, we'd still have the half-a-wolf problem once the new mount art gets merged in.

Adding the heads a second time with correct cropping and positioning would bloat the size of the images, but create less of a headache for maintainability and creating new sprites in the future. (You'd have to edit the grunt file every time a new mount got added if going the pure css route). So that's my vote. 

@SabreCat, have any thoughts?

@lemoness If we decide to go the extra-mount-head-image route, let's hold off on adding a trello card until the existing mounts are done. I want to make sure it's doable and is actually an improvement before putting the call out.

**Comment by SabreCat** [comment]
A head-only image might be nice to have; I like that solution. I wonder a bit about file size, but I do recall @lefnire saying at one point that image loading is thus far a relatively minor consideration with respect to performance.

**Comment by lefnire** [comment]
Perf is all user.batchUpdate(). 99%

**Comment by TheHollidayInn** [comment]
Hello,

I have been working with this ticket and #4803 . I have been able to align the mounts by doing the following:
1. Adding bootstrap column class to the li.customize-menu under the .pets div
2. Adding bootstrap column class to the div template for the actual mount 
3. Using a text-align: center property and value pair on the div above
4. Adding margin:auto property value pair on the button that holds the mount image.

Now, this almost completes the request above, except for the image is not centered on the button. This can be solved two ways:
1. Center the image using css (there seems to be a class per a mount)
2. Center and align the sprites on the sprite sheet (which would also require a mandate of max proportions on pixel art for mounts.)

Does this sound like the correct direction? And, does anyone have input if so.

**Comment by lemoness** [comment]
So, we already have enough completed mount art for the next two years, so
changing the proportions now would be a massive hassle that would require
redoing a lot of the art. Also, wouldn't changing the mount CSS alter where
avatars are sitting on the mounts?

**Comment by crookedneighbor** [comment]
@lemoness Not if the change was made specific to the stable page, ie: `.stable-page-class .wolf-class`, then it would just affect the sprites in the stable page.

**Comment by joyheron** [comment]
Just want to mention that this is same issue as in #2084.

**Comment by joyheron** [comment]
I think that the alignment of most of the mount heads could be improved with css magic. You could just copy the css class for the head image to a class .mount-class-centered and manipulate the background position numbers so that the heads are then aligned. I'm thinking of playing around with the idea, and if it looks good, doing a pull request.

**Comment by crookedneighbor** [comment]
@joyclark Yes, that's one route to go. The big problem right now is that the mount images are not consistent. Some mounts are just the head, and some are parts of the head and the body. It's done that way so that the avatar's are sitting on the right spot on the mount.

If we decide to create custom images for the mount selection page, whether it's mount heads or mini versions of the mounts, this problem will be solved because each will be the same dimensions and centered.

But, if we decide to live with it as is, manipulation the positioning of the heads via css is definitely the route to go. And if you do that, create a new `.styl` file in `website/public/css` to do it. That way, we can easily remove the styles if we decide to eventually replace them with unique images. 

Does that make sense?

**Comment by crookedneighbor** [comment]
@joyclark if you want to adjust a few, say 3, and submit a PR, I'll be happy to check your work and make sure it's on the right track.

**Comment by joyheron** [comment]
Sounds good. =)

**Comment by lemoness** [comment]
Note that we have a ton of completely redone art for all the mounts save a
few of the wolves, which are upcoming, so I am not sure this is a good use
of time.

**Comment by deilann** [comment]
All of the mounts?

Including the quest mounts? If the quest mounts aren't included, they could be worked on first.

I mean, the T-Rex mounts are so awful that they take up two rows:
![image](https://cloud.githubusercontent.com/assets/6274734/7572357/53415060-f7d2-11e4-85f9-192891532aa1.png)

**Comment by lemoness** [comment]
Nope, just base mounts - so you could definitely start with quest mounts!
But I wouldn't touch the base mounts until that art is changed.

**Comment by joyheron** [comment]
Ok, so I just saw this and already did an experiment for the first five base mounts. Whoops! Just for reference, here are before and after screenshots:

Before:
![uneven-mounts](https://cloud.githubusercontent.com/assets/1502724/7573264/d78deeb2-f824-11e4-88bb-1d0fdc757964.png)

After:
![mounts-aligned](https://cloud.githubusercontent.com/assets/1502724/7573270/e0d6db6e-f824-11e4-9e2a-63d93a6f9000.png)

I still went ahead and did a pull request, because I figured I would try to fix the quest mounts in the same way.

**Comment by joyheron** [comment]
Essentially, I figured out how wide each of the mount heads themselves were, and edited the gruntfile.js so that when sprites are generated a second class .Mount_Head_Type-centered is generated which differs in the width of the sprite image (I set the width to the width of the actual head image) and then added a margin to either side so that the whole class fills the desired 105px.

If this is a good idea (and I really don't know if it is), I could go ahead and do the same for the quest mounts (and remove the styling from the normal mounts if we want to wait for the new artwork).

The skeleton heads are still a bit off because they are thinner than the other head images of their type.

**Comment by lemoness** [comment]
If you've already done it, no need to revert - I just wanted to save you
time :)

**Comment by joyheron** [comment]
It doesn't actually take that much time. Probably 10 minutes maximum per mount type? It is just a matter of figuring out how wide the mount image is and then the margin size is (105 - head-width)/2. Then just update the file, generate the sprites, and reload to see the results. 

It actually takes more time to hatch and saddle all of the mounts for testing purposes.

**Comment by joyheron** [comment]
It seems to work for TRex's and Penguins as well...
![trex-after](https://cloud.githubusercontent.com/assets/1502724/7574070/da3ac968-f82a-11e4-9d00-7233ff284c2e.png)

**Comment by hairlessbear** [comment]
The new wolf mount art is wider than it used to be, causing the wolf mounts to be out of alignment with the rest of the base mounts (which are aligned properly thanks to @joyclark).

![selection_013](https://cloud.githubusercontent.com/assets/7059890/8435739/034e17de-1f21-11e5-9a1f-ea80c2edbc3a.jpg)

**Comment by lemoness** [comment]
So, we're going to switch over to displaying separate art of the isolated
mount heads in a fixed canvas - the art is currently being generated on the
Pixel Art Trello. I'll post them when they are done! That will fix all
mount page alignment issues.

**Comment by hairlessbear** [comment]
Yay!

**Comment by zelq** [comment]
I cannot express how much my OCD thanks you! (dusted off an old github account just to say this!)

**Comment by MaidM** [comment]
Further to the 30 June report on the wolf mount art throwing everything out of alignment, are we any closer to a fix on this?  It's been driving me mildly nuts for months.  Frankly, I'd rather go back to the old style wolf mount heads than have the new more detailed art misaligned to everything else.  Keep the same style everywhere, and if you're changing it, wait until they're all ready to change (and also until the artwork doesn't force a wrap to a second line the way the wolves currently do at my max resolution).

![image](https://cloud.githubusercontent.com/assets/11985555/10579458/165c4ade-7670-11e5-820d-87e0fd6ab488.png)

**Comment by crookedneighbor** [comment]
@MaidM It requires pixel art manipulations. It is currently being worked on.

@lemoness Any update on how many are left to do?

**Comment by lemoness** [comment]
Waiting on a total of 10, with 26 completed.

Of the ones that are currently/about to be in the game, we are missing
seven.

Is this something that can be done piecemeal, or does it need to be all at
once?

**Comment by crookedneighbor** [comment]
It can be done a little at a time.

**Comment by lemoness** [comment]
The art for the currently completed Mount Heads is found on this Trello
card:

https://trello.com/c/8b70B8V6/36-mount-heads

I will post here as more are completed.

The ones that can currently be implemented are:

Cobras - @Rbrinkley 

Roosters - @rbrinkley

Whales - @Rbrinkley 

Gryphons - @rbrinkley

Tigers - @zorella

Rare Mounts (Polar Bear, Mantis Shrimp, Turkey, Mammoth, and Orca) -
@zorella

Wolves - @rbrinkley

Bears - @zorella

Cheetah - @rbrinkley

Pandas - @leafybear

Horse - @rbrinkley

Rock - @zorella

LIons - @zorella

Mounts for Egg Pets - @zorella

Bunny - @zorella

Marshmallow Slime - @zorella

Deer - @rbrinkley

Hedgehog - @rbrinkley

Foxes - @zorella

Octopus - @rbrinkley

Rats - @rbrinkley

T-Rexes @eevachu

Monkey - @nonight

Spider Mounts @eevachu

Seahorse @eevachu

Penguin @eevachu

Parrot - @eevachu

Sheep - @nonight

Owl @eevachu

**Comment by lemoness** [comment]
Rooster and Whale mount heads are complete as well! They can be found on the Trello card. I'm editing my previous comment to be up-to-date.

**Comment by lemoness** [comment]
Cobra heads are done! They can be found on the Trello card. I'm editing my previous comment to be up-to-date.

**Comment by lemoness** [comment]
With the exception of the spooky mount heads, all of the art is done! https://trello.com/c/8b70B8V6/36-mount-heads

**Comment by joyheron** [comment]
The artwork looks really good! 
Has anyone made a pull request to implement the coding side of this yet? If not, I might give it a stab this weekend. 
I don't know exactly every task that needs to get done, but I think the following tasks need to be accomplished:
- Uploading the finished artwork into the repository (renaming the images systematically)
- Check sprite generation and generate css files
- Update mount stable page so that it uses the correct css. I assume we should still have the default mount heads if the new mount heads are not available (for instance for the spooky mount heads).

A question to consider would be how to name the new css classes, since the Mount_Head_ prefix has already been taken.

**Comment by crookedneighbor** [comment]
@joyclark go for it.

Regarding the naming scheme, I was thinking `Mount_Stable_Head_<whatever>`

**Comment by lemoness** [comment]
Another could be Mount_Icon, if you want it to be shorter!

**Comment by crookedneighbor** [comment]
I like Mount_Icon!

**Comment by joyheron** [comment]
I am working on this right now, but I just wanted to mention that there is some artwork which is currently not done:
- the spooky heads (this was already mentioned)
- the phoenix
- some of the gryphons (Red, RoyalPurple, Shade, Skeleton, White, Zombie)
- the frogs

This doesn't mean that they haven't been completed, but I couldn't find the images on the Trello card.

My plan is to copy the existing Mount_Head_ to be the Mount_Icon_ for these ones. Then, when the completed image has been created or found, it should just be a simple matter of inserting the correct image file.

**Comment by joyheron** [comment]
Oh, and I also could not find the JackOLantern mount

**Comment by lemoness** [comment]
Thanks for bringing those up! I will add them to the list of needs on that
Trello card.

**Comment by lemoness** [comment]
I've also added a call for the new Peppermint Mount heads.

**Comment by joyheron** [comment]
Also for future reference, if the images are already labelled as: Mount_Icon_{Animal}-{Color}.png by the pixel artists, that would save some time in the future (it would mean that the coders (or whoever is uploading the images into the repository) don't have to rename every file).

**Comment by crookedneighbor** [comment]
@joyclark's work has been merged in. 

Still needed:
- [ ] frogs
- [ ] phoenix
- [ ] jackolantern

I also noticed that many of the mount icons did not have transparent backgrounds :disappointed: Will work for now, but in the long haul, that should be fixed.
- [ ] flying pigs
- [ ] cactus
- [ ] cuttlefish 
- [ ] dragon

**Comment by lemoness** [comment]
It turns out that files uploaded with dropbox seem to lose their
transparent backgrounds :( I'm talking to the artists about it.

**Comment by lemoness** [comment]
Transparent backgrounds are done, and so are frogs, phoenix, and jackolantern! https://trello.com/c/8b70B8V6/36-mount-heads

Peppermint mount heads and Spooky mount heads are still in progress.

**Comment by crookedneighbor** [comment]
@lemoness I did the peppermint and spooky mount heads already :)

**Comment by lemoness** [comment]
Aha! In that case, we should be set. I will notify the artist :)

**Comment by librarianmage** [comment]
What is the status on this issue @Alys?

**Comment by Alys** [comment]
It looks like the frogs, phoenix, and jackolantern still need to be implemented, with the images being available in the link from lemoness a few comments up.

So, help is welcome now from a coder to fix this.

**Comment by librarianmage** [comment]
@Alys I can do this

**Comment by librarianmage** [comment]
@Alys My current understanding of this issue is:
- [ ] Upload all mount images that haven't yet
- [ ] Rename the images to `Mount_Icon_{Animal}-{Color}.png`
- [ ] Compile sprite images

Is this correct?

**Comment by Alys** [comment]
@MathWhiz That sounds correct to me.

**Comment by librarianmage** [comment]
@Alys How do I run the `grant-all-mounts` script?

**Comment by crookedneighbor** [comment]
if you run 

``` bash
node debug-scripts/grant-all-mounts.js
```

It'll tell you the usage, which is to supply the user id as the final argument

```
node debug-scripts/grant-all-mounts.js fill-this-in-with-your-local-user-id
```

**Comment by librarianmage** [comment]
@crookedneighbor Thanks!

**Comment by librarianmage** [comment]
@crookedneighbor Do I need to have `npm start` running when I run `npm run sprites`?

**Comment by librarianmage** [comment]
Frogs:
![screenshot 2016-04-20 at 5 32 11 pm](https://cloud.githubusercontent.com/assets/12386291/14692601/e08e8806-071d-11e6-83be-c7923b9a5bbe.png)

**Comment by librarianmage** [comment]
Rare Mounts:
![screenshot 2016-04-20 at 5 47 44 pm](https://cloud.githubusercontent.com/assets/12386291/14692890/122dfb9c-0720-11e6-8c09-f8e3042f090f.png)

**Comment by lemoness** [comment]
This looks great! Thank you.

**Comment by librarianmage** [comment]
@lemoness I'm glad you like it!
